libs/graph_utils/src/iso.rs

```diff
@@ -536,20 +623,33 @@ mod tests {
 
     #[test]
     fn test_shamir_on_bookinfo() {
-        let graph_g = bookinfo_trace_graph();
-        let graph_h = three_node_graph();
-        let mapping_wrapped = find_mapping_shamir_centralized(&graph_g, &graph_h);
+        let mut graph_g = bookinfo_trace_graph();
+        let mut graph_h = three_node_graph();
+        let mut mapping_wrapped = find_mapping_shamir_centralized(&graph_g, &graph_h);
         assert!(mapping_wrapped.is_some());
-        let mapping = mapping_wrapped.unwrap();
-        let a = get_node_with_id(&graph_h, "a".to_string()).unwrap();
-        let b = get_node_with_id(&graph_h, "b".to_string()).unwrap();
-        let c = get_node_with_id(&graph_h, "c".to_string()).unwrap();
-        let prod = get_node_with_id(&graph_g, "productpage-v1".to_string()).unwrap();
-        let det = get_node_with_id(&graph_g, "details-v1".to_string()).unwrap();
-        let rev = get_node_with_id(&graph_g, "reviews-v1".to_string()).unwrap();
+        let mut mapping = mapping_wrapped.unwrap();
+        let mut a = get_node_with_id(&graph_h, "a".to_string()).unwrap();
+        let mut b = get_node_with_id(&graph_h, "b".to_string()).unwrap();
+        let mut c = get_node_with_id(&graph_h, "c".to_string()).unwrap();
+        let mut prod = get_node_with_id(&graph_g, "productpage-v1".to_string()).unwrap();
+        let mut det = get_node_with_id(&graph_g, "details-v1".to_string()).unwrap();
+        let mut rev = get_node_with_id(&graph_g, "reviews-v1".to_string()).unwrap();
```

Comment:
Why does all of this have to be mutable? I am assuming it is because you are overwriting this variable later?

Reply:
Yes, there's two testing conditions in that test.  First, we test whether bookinfo and a three node graph match when the three node graph is a parent with two children.  Then we test whether bookinfo and a three node graph match when the three node graph is a chain.

Reply:
Why not create new variables for this? Also reduces the possibility of making mistakes.

Reply:
That makes sense.  Should be distinct variables now.